interface/modules/custom_modules/oe-module-faxsms/src/Controller/AppDispatch.php

```diff
@@ -0,0 +1,537 @@
+    }
+
+    /**
+     * @param $param
+     * @param $default
+     * @return mixed|null
+     */
+    public function getRequest($param = null, $default = null): mixed
+    {
+        if ($param) {
+            return $this->_request[$param] ?? $default;
+        }
+
+        return $this->_request;
+    }
+
+    /**
+     * @return mixed
+     */
+    static function getModuleVendor(): mixed
+    {
+        switch ((string)self::getServiceType()) {
+            case '1':
+                return '_ringcentral';
+            case '2':
+                return '_twilio';
+            case '3':
+                return '_etherfax';
+        }
+        return null;
+    }
+
+    /**
+     * Common credentials storage between services
+     * the service class will set specific credential.
+     *
+     * @return array|mixed
+     */
+    protected function getSetup(): mixed
+    {
+        $vendor = self::getModuleVendor();
+        $this->authUser = (int)$this->getSession('authUserID');
+        if (!($GLOBALS['oerestrict_users'] ?? null)) {
+            $this->authUser = 0;
+        }
+        $credentials = sqlQuery("SELECT * FROM `module_faxsms_credentials` WHERE `auth_user` = ? AND `vendor` = ?", array($this->authUser, $vendor))['credentials'];
+
+        if (empty($credentials)) {
+            $credentials = array(
+                'username' => '',
+                'extension' => '',
+                'password' => '',
+                'account' => '',
+                'phone' => '',
+                'appKey' => '',
+                'appSecret' => '',
+                'server' => '',
+                'portal' => '',
+                'smsNumber' => '',
+                'production' => '',
+                'redirect_url' => '',
+                'smsHours' => "50",
+                'smsMessage' => "A courtesy reminder for ***NAME*** \r\nFor the appointment scheduled on: ***DATE*** At: ***STARTTIME*** Until: ***ENDTIME*** \r\nWith: ***PROVIDER*** Of: ***ORG***\r\nPlease call if unable to attend.",
+            );
+            return $credentials;
+        }
+
+        return json_decode($this->crypto->decryptStandard($credentials), true);
+    }
+
+    /**
+     * @return array
+     */
+    public static function getLoggedInUser(): array
+    {
+        $id = $_SESSION['authUserID'] ?? 1;
+        $query = "SELECT fname, lname, fax, facility FROM users WHERE id = ?";
+        $result = sqlQuery($query, array($id));
+
+        return $result;
+    }
+
+    /**
+     * @return string|false
+     */
+    public function getPatientDetails(): bool|string
+    {
+        $id = $this->getRequest('pid');
+        $query = "SELECT fname, lname, phone_cell FROM Patient_data WHERE pid = ?";
+        $result = sqlQuery($query, array($id));
+
+        return json_encode($result);
+    }
+
+    public function chartDocument(): string
+    {
+        $pid = $this->getRequest('pid');
+        $docid = $this->getRequest('docid');
```

Comment:
Is this document already stored as a document and then stored again as a document?

Reply:
it's a a diff id besides this isn't used yet.